Source/mpqapi.cpp

```diff
@@ -385,47 +419,45 @@ void AllocBlock(uint32_t blockOffset, uint32_t blockSize)
 
 int FindFreeBlock(uint32_t size, uint32_t *blockSize)
 {
-	int result;
-
-	_BLOCKENTRY *pBlockTbl = cur_archive.sgpBlockTbl;
-	for (int i = 0; i < INDEX_ENTRIES; i++, pBlockTbl++) {
-		if (pBlockTbl->offset == 0)
+	for (int i = 0; i < INDEX_ENTRIES; i++) {

```

Comment:
This can use a range based loop
```suggestion
	for (BlockEntry &block : cur_archive.blockTbl) {
```

`i` is not used in the loop beyond the first index, `block` is always dereferenced so doesn't need to be a pointer.